content/learn/05.communication/07.modbus/modbus.md

```diff
@@ -0,0 +1,119 @@
+
+When using the [Modbus library](https://www.arduino.cc/reference/en/libraries/arduinomodbus/) sending messages is fairly straightforward as you can see in the examplary request format function below:
+
+| Device Address |  Function Code  | Starting Register |  Register Count | 
+|   -----------  |   -----------   |    -----------    |   -----------   |
+|       0x21     | INPUT REGISTERS |       30107       |        2        |
+
+***Note that this request form is specific to the [Modbus library](https://www.arduino.cc/reference/en/libraries/arduinomodbus/) and only works with boards compatible with this library. Make sure to check the specifications for the board you are using!***
+
+You have to check the device-specific documentation to attain the right address, function code, starting registers and register count. The CRC error check is taken care of by the Modbus library. To make it easier to understand the example below shows how to use the Modbus library.
+
+### Example
+Let's say you have a Modbus-compatible energy meter working with RS-485. In our case, we use a model from [Finder](https://cdn.findernet.com/app/uploads/2021/09/20090052/Modbus-7M24-7M38_v2_30062021.pdf) but you can use a different one.
+
+Take a look at the request frame inside the Finder documentation:
+
+![](assets/finderRequestFrame.png)
+
+We can see that the device address is 21 (or 0x21).
+
+The function code states 04 which we know stands for INPUT REGISTERS ([see Function Code](#function-code)).
+
+The starting register is 00 6B (or 0x6B) but for the Modbus library to work we need to change it from hexadecimal to decimal.
+In this example, the starting register is stated as hexadecimal but if we scroll down further in the documentation we can see that the registers are written in decimals: 
+
+![](assets/modbusMeasurmentFinder.png)
+
+We can also see that the register for U1 takes up 30107 and 30108 which means that the register count is 2.
+
+With all this information we can now use the function requestFrom() inside the readVoltage() function:
+
+```
+float readVoltage() {
+  float volt = 0.;
+  // Send reading request over RS485 
+  if (!ModbusRTUClient.requestFrom(0x21, INPUT_REGISTERS, 30017, 2)) {
+    // Error handling
+    Serial.print("- Failed to read the voltage! ");
+    Serial.println(ModbusRTUClient.lastError()); 
+  } else {
+    // Response handler 
+    uint16_t word1 = ModbusRTUClient.read();  // Read word1 from buffer
+    uint16_t word2 = ModbusRTUClient.read();  // Read word2 from buffer
+    uint32_t millivolt = word1 << 16 | word2; // Join word1 and word2 to retrieve voltage value in millivolts
+    volt = millivolt/1000.0;                  // Convert to volts
+  }
+
+  return volt;
+}
+```
+
+
```

Comment:
For the following sections: Modbus compatible boards, Libraries, and Read more; it could be nice to have a little more description for the links and content. For example, mentioning: " These are the libraries that you can use to enable Modbus protocol capability found within Arduino boards .. " for the Libraries section and include this type of description for the other sections previously mentioned.